html/user/per_app_list.php

```diff
@@ -26,104 +26,243 @@
+                }
+                $y++;
             }
         }
-        echo "<td align=right>".format_credit($c->expavg)."</td><td align=right>".format_credit_large($c->total)."</td>\n";
+        
+        $x++;
     }
-    echo "</tr>\n";
+    return $store;
 }
 
-function show_list($is_team, $appid, $is_total) {
-    $x = $is_team?"teams":"participants";
-    page_head("Top $x by application");
-    $apps = BoincApp::enum("deprecated=0");
-    if (!$appid) {
-        $appid = $apps[0]->id;
+function retrieve_credit_user($data)
+{
+    global $apps;
+    $x = 1;
+    $c = 0;
+    
+    foreach ($data as $item) {
+        $user = BoincUser::lookup_id($item->userid);
+        if (is_object($user)) {
+            $sign         = user_links($user, BADGE_HEIGHT_MEDIUM);
+            $store[0][$c] = $sign;
+            $c++;
+        }
+    }
+    
+    foreach ($apps as $app) {
+        $y = 0;
+        
+        foreach ($data as $item) {
+            $user = BoincUser::lookup_id($item->userid);
+            if (is_object($user)) {
+                $item = BoincCreditUser::lookup("userid=$item->userid and appid=$app->id");
+                if (is_object($item)) {
+                    $z                 = 0;
+                    $store[$x][$y][$z] = $item->expavg;
+                    $z++;
+                    $store[$x][$y][$z] = $item->total;
+                } else {
+                    $z                 = 0;
+                    $store[$x][$y][$z] = 0.0;
+                    $z++;
+                    $store[$x][$y][$z] = 0.0;
+                }
+                $y++;
+            }
+        }
+        $x++;
     }
-    start_table('table-striped');
-    show_header($is_team, $apps, $appid, $is_total);
-    $x = $is_total?"total":"expavg";
+    return $store;
+}
+
+function get_top_items($is_team, $appid, $is_total, $offset)
+{
+    global $items_per_page;
+    $x = $is_total ? "total" : "expavg";
+    
     if ($is_team) {
-        $items = BoincCreditTeam::enum("appid=$appid order by $x desc");
+        
+        $data  = BoincCreditTeam::show_list("appid=$appid", $x, $offset . ", " . $items_per_page);
+        $store = retrieve_credit_team($data);
+        
     } else {
-        $items = BoincCreditUser::enum("appid=$appid order by $x desc");
+        $data  = BoincCreditUser::show_list("appid=$appid", $x, $offset . ", " . $items_per_page);
+        $store = retrieve_credit_user($data);
     }
-    $i = 0;
-    foreach ($items as $item) {
-        show_row($item, $apps, $is_team, $i);
-        $i++;
+    
+    return $store;
+}
+
+$is_team        = get_int('is_team', true);
+$appid          = get_int('appid', true);
+$is_total       = get_int('is_total', true);
+$items_per_page = 20;
+$offset         = get_int('offset', true);
+
+if (!$offset)
+    $offset = 0;
+if ($offset % $items_per_page)
+    $offset = 0;
+
+$x = $is_team ? "teams" : "participants";
+page_head(tra("Top %1 by application", $x));
```

Comment:
To give the translator a chance to translate this better I guess there should be two complete strings without the `%1`. like this:
```
$title= $is_team ? tra("Top teams by application") : tra("Top participants by application");
page_head($title);
```